flash-card.js

```diff
@@ -69,22 +85,64 @@ export default class FlashCardApp extends React.Component {
     for (var pair of formData.entries()) {
       cardObj[pair[0]] = pair[1]
     }
-    const flashCardStateCopy = this.state.flashcards.slice(0)
-    flashCardStateCopy.splice(editIndex, 1, cardObj)
+    const flashcardsCopy = this.state.flashcards.slice(0)
+    flashcardsCopy.splice(editIndex, 1, cardObj)
+    const updatedTopics= []
+    flashcardsCopy.map(flashcard =>  {
+      if (updatedTopics.indexOf(flashcard.topic) === -1) {
+        updatedTopics.push(flashcard.topic)
+      }
+    })
     this.setState({
       view: "Cards",
-      flashcards: flashCardStateCopy})
+      flashcards: flashcardsCopy,
+      topics: updatedTopics
+    })
   }
   handleDelete(event) {
     const index = event.target.getAttribute('data-index')
-    const flashCardStateCopy = this.state.flashcards.slice(0)
-    flashCardStateCopy.splice(index, 1)
+    const flashcardStateCopy = this.state.flashcards.slice(0)
+    flashcardStateCopy.splice(index, 1)
+    const updatedTopics= []
+    flashcardStateCopy.map(flashcard =>  {
+      if (updatedTopics.indexOf(flashcard.topic) === -1) {
+        updatedTopics.push(flashcard.topic)
+      }
+    })
+    this.setState({
+      flashcards: flashcardStateCopy,
+      topics: updatedTopics
+    })
+  }
+  handleSelectedTopic(event) {
+    const {flashcards} = this.state
+    const topic = event.target.textContent
+    const flashcardsFilteredCopy = this.state.flashcards.slice(0)
+    const filteredCards = flashcardsFilteredCopy.filter(flashcard => flashcard.topic === topic)
+    const updatedTopics = []
+    flashcards.map(flashcard => {
+      if (updatedTopics.indexOf(flashcard.topic) === -1) {
+        updatedTopics.push(flashcard.topic)
+      }
+    })
     this.setState({
-      flashcards: flashCardStateCopy
+      view: 'Practice',
+      selectedTopic: topic,
+      flashcardsFiltered: filteredCards,
```

Comment:
Instead of storing the filtered flash cards, just pass the `selectedTopic` as a prop to your list component. It can filter the cards on the fly when it goes to render.